src/mono/mono/mini/mini-runtime.c

```diff
@@ -2668,8 +2668,16 @@ gpointer
 mono_jit_compile_method (MonoMethod *method, MonoError *error)
 {
 	gpointer code;
+	MonoLMFExt ext;
+
+	memset (&ext, 0, sizeof (MonoLMFExt));
+	ext.kind = MONO_LMFEXT_JIT_ENTRY;
+	mono_push_lmf (&ext);
 
 	code = mono_jit_compile_method_with_opt (method, mono_get_optimizations_for_method (method, default_opt), FALSE, error);
+
+	mono_pop_lmf (&ext.lmf);
```

Comment:
Note that this will not be called if a c++ exception is thrown in llvmonly mode. Hopefully that is not going to happen in this code path, but to be safe, it might be good to wrap the mono_jit_compile_method_with_opt () call in mono_llvm_cpp_catch_exception () as done in do_jit_call ().